benchmarks/pytest-based/bench_algos.py

```diff
@@ -225,6 +226,19 @@ def build_personalization_dict(pagerank_dict):
     return pers_dict
 
 
+# Used to return a function that calls the original function inside a try-except block
+# which is Useful to still save pytest-benchmark numbers if failure is correct behavior
+# for the function being tested.
+def possible_to_fail(exception, function):
+    def nested_func(*args, **kwargs):
+        try:
+            return function(*args, **kwargs)
+        except exception:
+            pass
```

Comment:
Hmm, should we print or warn a message here and show the exception that's being squashed?